media/whep_server.go

```diff
@@ -0,0 +1,332 @@
+	}
+	if err := mpegtsReader.Initialize(); err != nil {
+		clog.InfofErr(ctx, "Failed to initialize mpegts reader", err)
+		http.Error(w, "Failed to initialize mpegts reader", http.StatusInternalServerError)
+		return
+	}
+	tracks := mpegtsReader.Tracks()
+	hasAudio, hasVideo, hasIDR := false, false, false
+	trackCodecs := make([]string, 0, len(tracks))
+	for _, track := range tracks {
+		switch track.Codec.(type) {
+		case *mpegts.CodecH264:
+			// TODO this track can probably be reused for multiple viewers
+			webrtcTrack, err := NewLocalVideoTrack(
+				webrtc.RTPCodecCapability{MimeType: webrtc.MimeTypeH264},
+				"video", "livepeer",
+			)
+			if err != nil {
+				clog.InfofErr(ctx, "Error creating track for h264", err)
+				http.Error(w, "Error creating track for h264", http.StatusInternalServerError)
+				return
+			}
+			// TODO use RTPSender.ReplaceTrack on orch swaps
+			if _, err := peerConnection.AddTrack(webrtcTrack); err != nil {
+				clog.InfofErr(ctx, "Error adding track for video", err)
+				http.Error(w, "Error adding track for h264", http.StatusInternalServerError)
+				return
+			}
+			mpegtsReader.OnDataH264(track, func(pts, dts int64, au [][]byte) error {
+				if !hasIDR {
+					if h264.IsRandomAccess(au) {
+						hasIDR = true
+					} else {
+						// drop until next keyframe
+						return nil
+					}
+				}
+				return webrtcTrack.WriteVideo(au, pts)
+			})
+			trackCodecs = append(trackCodecs, "h264")
+			hasVideo = true
+
+		case *mpegts.CodecOpus:
+			webrtcTrack, err := NewLocalVideoTrack(
+				// NB: Don't signal sample rate or channel count here. Leave empty to use defaults.
+				// Opus RTP RFC 7587 requires opus/48000/2 regardless of the actual content
+				webrtc.RTPCodecCapability{MimeType: webrtc.MimeTypeOpus},
+				"audio", "livepeer", // TODO can be another MediaID to desync a/v for latency
+			)
+			if err != nil {
+				clog.InfofErr(ctx, "Error creating track for opus", err)
+				http.Error(w, "Error creating track for opus", http.StatusInternalServerError)
+				return
+			}
+			// TODO use RTPSender.ReplaceTrack on orch swaps
+			if _, err := peerConnection.AddTrack(webrtcTrack); err != nil {
+				clog.InfofErr(ctx, "Error adding track for audio", err)
+				http.Error(w, "Error adding track for audio", http.StatusInternalServerError)
+				return
+			}
+			mpegtsReader.OnDataOpus(track, func(pts int64, packets [][]byte) error {
+				pts = multiplyAndDivide(pts, 48_000, 90_000)
+				return webrtcTrack.WriteVideo(packets, pts)
+			})
+			trackCodecs = append(trackCodecs, "opus")
+			hasAudio = true
+		}
+	}
+
+	if !hasAudio && !hasVideo {
+		clog.InfofErr(ctx, "No audio or video in media stream", errors.New("no audio or video"))
+		http.Error(w, "No audio or video in media stream", http.StatusInternalServerError)
+		return
+	} else if !hasVideo {
+		clog.Info(ctx, "No video in output")
+	} else if !hasAudio {
+		clog.Info(ctx, "No audio in output")
+	}
+	clog.Info(ctx, "Outputs", " tracks", trackCodecs)
```

Comment:
For now should we always err if `!hasVideo` and only log if no audio? Or maybe just log both props, together in the log below

```suggestion
	if !hasVideo {
		clog.InfofErr(ctx, "No video in media stream", errors.New("no video"))
		http.Error(w, "No video in media stream", http.StatusInternalServerError)
		return
	}
	clog.Info(ctx, "Outputs", "hasVideo", hasVideo, "hasAudio, hasAudio, "tracks", trackCodecs)
```

Reply:
I'm inclined to keep it going for _output_ even if there's no video, but consolidated the logging in 556bd849ed9fbceb9463ade375ca395360024489